fables/parse.py

```diff
@@ -30,17 +31,36 @@
 FRACTION_OF_BLANK_HEADERS_ALLOWED = 0.5
 
 
-def sniff_delimiter(bytesio: IO[bytes]) -> str:
-    # TODO(Thomas: 03/06/2019): Assuming utf-8 for now. Later should consider
-    #                           use of chardet and/or cchardet to auto-detect
-    #                           encoding.
-    sample = bytesio.read(1024 * 4).decode(encoding="utf-8")
+def sniff_delimiter(bytesio: IO[bytes], encoding: str) -> str:
```

Comment:
[blocking] If `encoding` can be `None` the type annotation should be `Optional[str]`.